tests/warp-drive__schema-record/tests/reads/alias-test.ts

```diff
@@ -0,0 +1,116 @@
+import { module, test } from 'qunit';
+
+import { setupTest } from 'ember-qunit';
+
+import { recordIdentifierFor } from '@ember-data/store';
+import type { StableRecordIdentifier } from '@warp-drive/core-types';
+import { Type } from '@warp-drive/core-types/symbols';
+import type { SchemaRecord } from '@warp-drive/schema-record/record';
+import type { Transformation } from '@warp-drive/schema-record/schema';
+import { registerDerivations, withDefaults } from '@warp-drive/schema-record/schema';
+
+import type Store from 'warp-drive__schema-record/services/store';
+
+interface User {
+  id: string | null;
+  $type: 'user';
+  rawNetWorth: string;
+  netWorth: number;
+  [Type]: 'user';
+}
+
+module('Reads | Alias fields', function (hooks) {
+  setupTest(hooks);
+
+  test('we can use simple fields with a `type`', function (assert) {
+    const store = this.owner.lookup('service:store') as Store;
+    const { schema } = store;
+
+    const FloatTransform: Transformation<string | number, number> = {
+      serialize(value: string | number, options: { precision?: number } | null, _record: SchemaRecord): string {
+        return typeof value === 'number'
+          ? value.toFixed(options?.precision ?? 3)
+          : Number(value).toFixed(options?.precision ?? 3);
+      },
+      hydrate(value: string, _options: { precision?: number } | null, _record: SchemaRecord): number {
+        if (value === undefined || value === null) {
+          return 0;
+        }
+        return Number(value);
+      },
+      defaultValue(_options: { precision?: number } | null, _identifier: StableRecordIdentifier): string {
+        const v = 0;
+        return v.toFixed(_options?.precision ?? 3);
+      },
+      [Type]: 'float',
+    };
+
+    schema.registerTransformation(FloatTransform);
+    registerDerivations(schema);
+
+    schema.registerResource(
+      withDefaults({
+        type: 'user',
+        fields: [
+          {
+            name: 'rawNetWorth',
+            kind: 'field',
+          },
+          {
+            kind: 'alias',
+            name: 'netWorth',
+            type: null,
+            options: {
+              name: 'rawNetWorth',
+              kind: 'field',
+              type: 'float',
+              options: { precision: 2 },
+            },
+          },
+        ],
+      })
+    );
+
+    const record = store.createRecord<User>('user', {
+      rawNetWorth: '1000000.009',
+    });
+    const identifier = recordIdentifierFor(record);
+    const resource = store.cache.peek(identifier)!;
+
+    assert.strictEqual(record.rawNetWorth, '1000000.009', 'netWorth is accessible in raw form');
+    assert.strictEqual(record.netWorth, 1_000_000.009, 'netWorth is accessible in numeric form');
```

Comment:
Hey I might be missing something, but I would expect `netWorth` to be 1_000_000.01 as it has transformer in alias setup

Reply:
hydration doesn't change the value in the transform we registered, only serialization does. Since we set the value here via the raw form, we don't expect serialization to run. This test is basically checking that set via the original property is unaffected.